### ISSUE TITLE
Quotes on property keys are removed from declaration file producing broken .d.ts 

### ISSUE THREAD

**Issue by nadavsinai** [microsoft/TypeScript#37866]
HI,
I'm upgrading a codebase from 3.5.3 to 3.8.3 and found this error, tsc does not produce any error on this file, but the declaration file emitted is broken and can not be consumed by another tsc compilation.

providing your confirmation (and maybe some direction) I'd be happy to file a PR myself.
Thanks

Nadav Sinai - Tech lead.
Philips-Algotec

**TypeScript Version:**  3.8.3
**notice this is a regression - Typescript 3.7 produces the correct code.**

**Search Terms:**

I searched issues & SO for  :
"declaration parse error quotes "
"TS1131" && "TS1131: Property or signature expected" (loads of irreverent stuff)
"ts1131 property or signature expected. quotes removed .d.ts"
**Expected behavior:**
Quotes should be retained.
**Actual behavior:**
Quotes are removed, broken declaration file is produced which is unparsable by tsc

**Related Issues:**
no
**Code**
```ts
export enum MouseButton {
	LEFT_BUTTON = 1,
	RIGHT_BUTTON = 2,
	MIDDLE_BUTTON = 4,
	XBUTTON1_BUTTON = 5,
	XBUTTON2_BUTTON = 6,
	NO_BUTTON = 0,
}

// tslint:disable-next-line:variable-name
export const DOMMouseButton = {
	'-1': MouseButton.NO_BUTTON,
	"0": MouseButton.LEFT_BUTTON,
	"1": MouseButton.MIDDLE_BUTTON,
	"2": MouseButton.RIGHT_BUTTON,
	"3": MouseButton.XBUTTON1_BUTTON,
	"4": MouseButton.XBUTTON2_BUTTON,
};

```

<details><summary><b>Declaration Output</b></summary>

```ts
export declare enum MouseButton {
    LEFT_BUTTON = 1,
    RIGHT_BUTTON = 2,
    MIDDLE_BUTTON = 4,
    XBUTTON1_BUTTON = 5,
    XBUTTON2_BUTTON = 6,
    NO_BUTTON = 0
}
export declare const DOMMouseButton: {
    -1: MouseButton;
    0: MouseButton;
    1: MouseButton;
    2: MouseButton;
    3: MouseButton;
    4: MouseButton;
};
```

</details>

<details><summary><b>Compiler Options</b></summary>

```json
{
  "compilerOptions": {
    "noImplicitAny": true,
    "strictNullChecks": true,
    "strictFunctionTypes": true,
    "strictPropertyInitialization": true,
    "strictBindCallApply": true,
    "noImplicitThis": true,
    "noImplicitReturns": true,
    "alwaysStrict": true,
    "esModuleInterop": true,
    "declaration": true,
    "experimentalDecorators": true,
    "emitDecoratorMetadata": true,
    "moduleResolution": 2,
    "target": "ES2017",
    "jsx": "React",
    "module": "ESNext"
  }
}
```

</details>

**Playground Link:** [Provided](https://www.typescriptlang.org/v2/en/play?#code/KYDwDg9gTgLgBMAdgVwLZwLIWQZ2AIWRhgkTgG8AoASABkBRAMQBUB9fAVWeYHkA5OAF44ARgA0NAEoBJAOIAJNp278hcAEwTqGaQBFdDdl14DhAFi0ANZSZFGVpuAFYrN-uvsm1ANi18enqrCAAwSAL6UlAD0UXAwOAA2AJaIMABcACZJOACGAEYJwAC0iKAwRcmlaQBuOVBJ+YUlOajAlKCQsHAAxqQ48Lo8GFi4BEQkZMJU1ADkRSIzaZjYeITEpAB0-oF8WgBEwXtLI6vjmwwsO-siR8ujaxMbOvqGbrs0e+q3J2PriBsyBRKYz8fYAZm+K1+j2sIL4dje+zMkPuZ3+sIcHkRlDCAG5KEA)

**Comment by a-tarasyuk** [comment]
It seems this issue related to https://github.com/microsoft/TypeScript/issues/37292.
